### PR TITLE
feat: Update module catalog related docs

### PR DESCRIPTION
**Pull request by Tomasz-Smelcerz-SAP**
**Description**
This PR updates the documentation with respect to recent changes in Module Catalog synchronization, introduced by #1851.

Changes proposed in this pull request:

- Update diagrams
- Update markdown docs

**Related issue(s)**
Fixes: #1851

### CODE REVIEW

File: docs/contributor/assets/kyma-operator-architecture.svg
[the remark this answers is not in the document]
Comment (Tomasz-Smelcerz-SAP) [reply docs/contributor/assets/kyma-operator-architecture.svg:1]: Thanks for spotting that! I don't know how, this object is there - when editing in `draw.io`. It's probably hidden or something.

File: docs/contributor/01-architecture.md [hunk 1]
```diff
@@ -20,9 +20,9 @@ To run, Lifecycle Manager uses the following workflow:
...
+3. To manage a module, Lifecycle Manager requires its definition and version related metadata. The definition and version related metadata for a module are represented by the ModuleTemplate and ModuleReleaseMeta CRs. ModuleTemplate CR represents a module in a particular version, while ModuleReleaseMeta describe mapping of module versions to available channels. All ModuleTemplate CRs along with their related ModuleReleaseMeta CRs exist in Kyma Control Plane which is the central cluster with Kyma infrastructure. The set of ModuleTemplate CRs and related ModuleReleaseMeta CRs available for a particular Kyma runtime is called the Module Catalog. Lifecycle Manager creates the Module Catalog based on labels, such as `internal`, or `beta`, and uses the synchronization mechanism to update the Module Catalog portfolio.
```
Comment (mmitoraj) [inline docs/contributor/01-architecture.md:23]:
```suggestion
3. To manage a module, Lifecycle Manager requires its definition and version-related metadata. The ModuleTemplate and ModuleReleaseMeta CRs represent the definition and version-related metadata for a module. The ModuleTemplate CR represents a module in a particular version, while the ModuleReleaseMeta CR describes the mapping between module versions and available channels. All ModuleTemplate CRs, along with their related ModuleReleaseMeta CRs, exist in Kyma Control Plane which is the central cluster with Kyma infrastructure. The set of ModuleTemplate CRs and related ModuleReleaseMeta CRs available for a particular Kyma runtime is called the Module Catalog. Lifecycle Manager creates the Module Catalog based on labels, such as `internal`, or `beta`, and uses the synchronization mechanism to update the Module Catalog portfolio.
```

File: docs/contributor/01-architecture.md [hunk 2]
```diff
@@ -20,9 +20,9 @@ To run, Lifecycle Manager uses the following workflow:
...
 
-4. Lifecycle Manager reads a ModuleTemplate CR and creates a Manifest CR. The Manifest CR represents resources that make up a module and are to be installed on a remote cluster by Lifecycle Manager.
+4. Lifecycle Manager uses ModuleRelease Meta and ModuleTemplate CRs to read module's definition and creates a Manifest CR. The Manifest CR represents resources that make up a module and are to be installed on a remote cluster by Lifecycle Manager.
```
Comment (mmitoraj) [inline docs/contributor/01-architecture.md:25]:
```suggestion
4. Lifecycle Manager uses ModuleReleaseMeta and ModuleTemplate CRs to read a module's definition and create a Manifest CR. The Manifest CR represents resources that make up a module and are to be installed on a remote cluster by Lifecycle Manager.
```

File: docs/contributor/resources/README.md [hunk 3]
```diff
@@ -10,7 +10,7 @@ The API of Lifecycle Manager is based on Kubernetes Custom Resource Definitions
 ## Synchronization of Module Catalog with Remote Clusters
 
 Lifecycle Manager ensures the Module Catalog is correctly synchronized with users' runtimes.
-The Module Catalog consists of all modules, represented by ModuleTemplates CR, that are available for a user. The Module Catalog portfolio may vary for different users.
+The Module Catalog consists of all modules, represented by ModuleTemplates CR and related ModuleReleaseMetas CR, that are available for a user. The Module Catalog portfolio may vary for different users.
```
Comment (mmitoraj) [inline docs/contributor/resources/README.md:13]:
```suggestion
The Module Catalog consists of all modules, represented by ModuleTemplate CRs and related ModuleReleaseMetas CRs, that are available for a user. The Module Catalog portfolio may vary for different users.
```

**Review by nesmabadr — APPROVED** [review]

**Pull request merged**